com/uber/kafkaSpraynozzle/KafkaSpraynozzle.java

```diff
@@ -105,37 +111,52 @@ public static void main(String[] args) {
+        if (statsReporter == null) {
+            statsReporter = new NoopStatsReporter();
+        }
+        System.out.println("Using stats reporter: " + statsReporter);
+        return statsReporter;
+    }
+
+    private static Object getClass(String classpath, String className, String classArgs){
+        Object newClass = null;
+        if (classpath != null && className != null) {
+            File file = new File(classpath);
             try {
                 URL[] urls = new URL[]{file.toURL()};
                 ClassLoader cl = new URLClassLoader(urls);
-                Class<?> cls = cl.loadClass(filterClass);
-                if (filterClassArgs == null) {
-                    messageFilter = (KafkaFilter) cls.newInstance();
+                Class<?> cls = cl.loadClass(className);
+                if (classArgs == null) {
+                    newClass = cls.newInstance();
                 } else {
-                    messageFilter = (KafkaFilter) cls.getConstructor(String.class).newInstance(filterClassArgs);
+                    newClass = cls.getConstructor(String.class).newInstance(classArgs);
                 }
             } catch (MalformedURLException e) {
-                System.out.println("Bad classpath provided");
+                System.out.println("Bad classpath provided: " + classpath);
             } catch (ClassNotFoundException e) {
-                System.out.println("Filter class not found");
+                System.out.println("Class not found: " + className);
             } catch (InstantiationException e) {
-                System.out.println("Cannot create instance of filter class");
+                System.out.println("Cannot create instance of class: " + className);
             } catch (IllegalAccessException e) {
-                System.out.println("Filter class did I have no idea what but its bad and illegal.");
+                System.out.println("Class did I have no idea what but its bad and illegal: " + className);
             } catch (NoSuchMethodException e) {
-                System.out.println("Filter with custom arguments must have a constructor taking 1 String");
+                System.out.println("Class with custom arguments must have a constructor taking 1 String: " + className);
             } catch (InvocationTargetException e) {
-                System.out.println("Error initializing custom filter: " + e.getMessage());
+                System.out.println("Error initializing custom class: " + className + " " + e.getMessage());
+            } catch (Exception e) {
+                System.out.println("Unknown error initializing custom class: " + className + " " + e.getMessage());
             }
         }
-
-        if (messageFilter == null) {
-            messageFilter = new KafkaNoopFilter();
-        }
-        System.out.println("Using message filter: " + messageFilter);
-        return messageFilter;
+        return newClass;
```

Comment:
The diff makes this a bit messy, but essentially we have 3 methods: getKafkaFilter and getStatsReporter which are thin and essentially just call getClass which has all the old code for creating a class based on path, filename and args.